### ISSUE TITLE
[BUG] Binance clock desync in Windows results to problems with orders

### ISSUE THREAD

**Issue by Howie620** [CoinAlpha/hummingbot#1388]
**Describe the bug**
// A clear and concise description of what the bug is.
When I was running the bot on Binance yesterday, I saw 2 bids and 1 offer outstanding in single order mode and I have to manually cancel an outstanding order. 

**Steps To Reproduce**
// A concise description of the **steps to reproduce the buggy behavior**:
After I run the bot for a while, I have this issue. 

Here are the steps to reproduce the issue (see attachments in the section below):
1. Set up your configuration file with these parameters: ...
2. Start the bot using this command: ...
... Here is my config file for reference: 

**Screenshots**
// If applicable, add screenshots to help explain your problem.
![IMG_5448](https://user-images.githubusercontent.com/1378301/73900452-c42b7f00-48ca-11ea-8469-1c8d0b94e949.PNG)
![IMG_5447](https://user-images.githubusercontent.com/1378301/73900458-c8f03300-48ca-11ea-9626-efb832807c47.PNG)
As you can see in the screenshot and look closer at the timestamp, I still have this order outstanding. The bot is running correctly as there are other orders executed recently. 

**Release version**
// Include your bot's version number (Can be found at the upper left corner of your CLI).
0.23.0

**Attachments**
// Please attach your config file and log file to help us debug. It would be difficult for us to help you without those! 

[conf_pure_market_making_strategy_zil_eth.yml.txt](https://github.com/CoinAlpha/hummingbot/files/4162913/conf_pure_market_making_strategy_zil_eth.yml.txt)
[logs_conf_pure_market_making_strategy_zil_eth.log.txt](https://github.com/CoinAlpha/hummingbot/files/4162915/logs_conf_pure_market_making_strategy_zil_eth.log.txt)

*WARNING: Do NOT publish any exchange API keys or your wallet's private key. Whoever has access to them may steal your assets!*

**Optional: your discord username:** Howie#7291
**Optional: your ETH address:** (If you want to participate in the bug bounty)

**Comment by Hashira-Lars** [comment]
I've sent you a dm @Howie620 to follow up things faster.we can see from the logs one of the errors your having is with timestamp.

```binance.exceptions.BinanceAPIException: APIError(code=-1021): Timestamp for this request was 1000ms ahead of the server's time.```

**Comment by Hashira-Lars** [comment]
Cannot reproduce behavior in my end running for 2days windows binary.

@znerol19619 can you update this on your findings i think maybe you have info for this as your running your bot for 5 days now. Thanks

**Comment by znerol19619** [comment]
I was able to reproduce the behavior this time, after running it for 5 days. Today is the first time i actually encountered it. 

![Screenshot from 2020-02-10 18-14-03](https://user-images.githubusercontent.com/27669951/74141832-9cbe1480-4c32-11ea-8947-5c83e748ea47.png)

**Comment by Howie620** [comment]
Ok thanks. From my experience, I had to manually cancel the order in Binance.

**Comment by dennisocana** [comment]
Performed some tests and ran into this issue.

Clock desync errors with Binance gets printed over and over resulting to failure fetching accounts updated. Orders are not being cancelled and replaced when this happens.

In the attached log file, this happened for about an hour before Hummingbot recovered.

```
2020-02-14 16:35:17,109 - 9188 - hummingbot.market.binance.binance_market - NETWORK - Unexpected error while fetching account updates.
Traceback (most recent call last):
  File "hummingbot\market\binance\binance_market.pyx", line 736, in hummingbot.market.binance.binance_market.BinanceMarket._status_polling_loop
  File "hummingbot\core\utils\async_utils.py", line 19, in safe_gather
  File "hummingbot\market\binance\binance_market.pyx", line 299, in _update_balances
  File "hummingbot\market\binance\binance_market.pyx", line 278, in query_api
  File "hummingbot\core\utils\async_call_scheduler.py", line 128, in call_async
  File "hummingbot\core\utils\async_call_scheduler.py", line 117, in schedule_async_call
  File "hummingbot\core\utils\async_call_scheduler.py", line 80, in _coro_scheduler
  File "concurrent\futures\thread.py", line 56, in run
  File "site-packages\binance\client.py", line 1563, in get_account
  File "site-packages\binance\client.py", line 207, in _get
  File "site-packages\binance\client.py", line 181, in _request_api
  File "site-packages\binance\client.py", line 176, in _request
  File "site-packages\binance\client.py", line 200, in _handle_response
binance.exceptions.BinanceAPIException: APIError(code=-1021): Timestamp for this request was 1000ms ahead of the server's time.
```

**Steps To Reproduce**

1. Install Hummingbot locally via binary from the website.
2. Configure and run a basic pure market making on any trading pair in Binance.
3. You may have to wait very long and keep the bot running for the errors to appear.


**Release version**

Version 0.23.0
Windows 10


**Attachments**

[conf_pmm_binance_one-usdc.txt](https://github.com/CoinAlpha/hummingbot/files/4208242/conf_pmm_binance_one-usdc.txt)
[win_cex_pmm_single.log](https://github.com/CoinAlpha/hummingbot/files/4208239/win_cex_pmm_single.log)

**Comment by martinkou** [comment]
@yzhang1994 @Nullably

**Comment by Nullably** [comment]
This current timestamp offset calculation is to average the last stored 100 time offsets (between the local bot and Binance, stored every 60 seconds). This results in the offset not accurate enough for the timestamp to stay within Binance window (1000 ms) if the bot has been running for a while. 

An easy method to reproduce this error is to manually update your system clock to a few seconds forward/backward after the bot started.

The PR fixes this.

**Comment by martinkou** [comment]
Fixed with #1444.

**Comment by dennisocana** [comment]
Fix will be included in tomorrow's 0.24.0 release.

@Howie620 may I have the ETH address where bug bounty reward will be sent?

**Comment by neohowie** [comment]
Here you go. Changed my name from @Howie620 to @neohowie 

0xFE15aEeB4162da4Be2Bee2E8b702454Bf46C4217
